File: Dummy/hawkshaw/library/FileUploadResponse.java
package com.hawkshaw.library;

import com.google.protobuf.C0346c;
import com.google.protobuf.C0366m;
import com.google.protobuf.C0374q;
import com.google.protobuf.C0382u0;
import com.google.protobuf.C0385x;
import com.google.protobuf.D;
import com.google.protobuf.E;
import com.google.protobuf.H;
import com.google.protobuf.I;
import com.google.protobuf.M;
import com.google.protobuf.N;
import com.google.protobuf.O;
import java.io.InputStream;
import java.nio.ByteBuffer;

public final class FileUploadResponse extends I implements FileUploadResponseOrBuilder {
    /* access modifiers changed from: private */
    public static final FileUploadResponse DEFAULT_INSTANCE;
    public static final int NAME_FIELD_NUMBER = 1;
    private static volatile C0382u0 PARSER = null;
    public static final int PUBLIC_URL_FIELD_NUMBER = 2;
    public static final int STATUS_FIELD_NUMBER = 3;
    private String name_ = "";
    private String publicUrl_ = "";
    private int status_;

    public static final class Builder extends D implements FileUploadResponseOrBuilder {
        public /* synthetic */ Builder(int i5) {
            this();
        }

        public Builder clearName() {
            copyOnWrite();
            ((FileUploadResponse) this.instance).clearName();
            return this;
        }

        public Builder clearPublicUrl() {
            copyOnWrite();
            ((FileUploadResponse) this.instance).clearPublicUrl();
            return this;
        }

        public Builder clearStatus() {
            copyOnWrite();
            ((FileUploadResponse) this.instance).clearStatus();
            return this;
        }

        public String getName() {
            return ((FileUploadResponse) this.instance).getName();
        }

        public C0366m getNameBytes() {
            return ((FileUploadResponse) this.instance).getNameBytes();
        }

        public String getPublicUrl() {
            return ((FileUploadResponse) this.instance).getPublicUrl();
        }

        public C0366m getPublicUrlBytes() {
            return ((FileUploadResponse) this.instance).getPublicUrlBytes();
        }

        public Status getStatus() {
            return ((FileUploadResponse) this.instance).getStatus();
        }

        public int getStatusValue() {
            return ((FileUploadResponse) this.instance).getStatusValue();
        }

        public Builder setName(String str) {
            copyOnWrite();
            ((FileUploadResponse) this.instance).setName(str);
            return this;
        }

        public Builder setNameBytes(C0366m mVar) {
            copyOnWrite();
            ((FileUploadResponse) this.instance).setNameBytes(mVar);
            return this;
        }

        public Builder setPublicUrl(String str) {
            copyOnWrite();
            ((FileUploadResponse) this.instance).setPublicUrl(str);
            return this;
        }

        public Builder setPublicUrlBytes(C0366m mVar) {
            copyOnWrite();
            ((FileUploadResponse) this.instance).setPublicUrlBytes(mVar);
            return this;
        }

        public Builder setStatus(Status status) {
            copyOnWrite();
            ((FileUploadResponse) this.instance).setStatus(status);
            return this;
        }

        public Builder setStatusValue(int i5) {
            copyOnWrite();
            ((FileUploadResponse) this.instance).setStatusValue(i5);
            return this;
        }

        private Builder() {
            super(FileUploadResponse.DEFAULT_INSTANCE);
        }
    }

    public enum Status implements M {
        PENDING(0),
        IN_PROGRESS(1),
        SUCCESS(2),
        FAILED(3),
        UNRECOGNIZED(-1);
        
        public static final int FAILED_VALUE = 3;
        public static final int IN_PROGRESS_VALUE = 1;
        public static final int PENDING_VALUE = 0;
        public static final int SUCCESS_VALUE = 2;
        private static final N internalValueMap = null;
        private final int value;

        /* JADX WARNING: type inference failed for: r0v6, types: [com.google.protobuf.N, java.lang.Object] */
        static {
            internalValueMap = new Object();
        }

        private Status(int i5) {
            this.value = i5;
        }

        public static Status forNumber(int i5) {
            if (i5 == 0) {
                return PENDING;
            }
            if (i5 == 1) {
                return IN_PROGRESS;
            }
            if (i5 == 2) {
                return SUCCESS;
            }
            if (i5 != 3) {
                return null;
            }
            return FAILED;
        }

        public static N internalGetValueMap() {
            return internalValueMap;
        }

        public static O internalGetVerifier() {
            return j.f5045a;
        }

        public final int getNumber() {
            if (this != UNRECOGNIZED) {
                return this.value;
            }
            throw new IllegalArgumentException("Can't get the number of an unknown enum value.");
        }

        @Deprecated
        public static Status valueOf(int i5) {
            return forNumber(i5);
        }
    }

    static {
        FileUploadResponse fileUploadResponse = new FileUploadResponse();
        DEFAULT_INSTANCE = fileUploadResponse;
        I.registerDefaultInstance(FileUploadResponse.class, fileUploadResponse);
    }

    private FileUploadResponse() {
    }

    /* access modifiers changed from: private */
    public void clearName() {
        this.name_ = getDefaultInstance().getName();
    }

    /* access modifiers changed from: private */
    public void clearPublicUrl() {
        this.publicUrl_ = getDefaultInstance().getPublicUrl();
    }

    /* access modifiers changed from: private */
    public void clearStatus() {
        this.status_ = 0;
    }

    public static FileUploadResponse getDefaultInstance() {
        return DEFAULT_INSTANCE;
    }

    public static Builder newBuilder() {
        return (Builder) DEFAULT_INSTANCE.createBuilder();
    }

    public static FileUploadResponse parseDelimitedFrom(InputStream inputStream) {
        return (FileUploadResponse) I.parseDelimitedFrom(DEFAULT_INSTANCE, inputStream);
    }

    public static FileUploadResponse parseFrom(ByteBuffer byteBuffer) {
        return (FileUploadResponse) I.parseFrom(DEFAULT_INSTANCE, byteBuffer);
    }

    public static C0382u0 parser() {
        return DEFAULT_INSTANCE.getParserForType();
    }

    /* access modifiers changed from: private */
    public void setName(String str) {
        str.getClass();
        this.name_ = str;
    }

    /* access modifiers changed from: private */
    public void setNameBytes(C0366m mVar) {
        C0346c.checkByteStringIsUtf8(mVar);
        this.name_ = mVar.r();
    }

    /* access modifiers changed from: private */
    public void setPublicUrl(String str) {
        str.getClass();
        this.publicUrl_ = str;
    }

    /* access modifiers changed from: private */
    public void setPublicUrlBytes(C0366m mVar) {
        C0346c.checkByteStringIsUtf8(mVar);
        this.publicUrl_ = mVar.r();
    }

    /* access modifiers changed from: private */
    public void setStatus(Status status) {
        this.status_ = status.getNumber();
    }

    /* access modifiers changed from: private */
    public void setStatusValue(int i5) {
        this.status_ = i5;
    }

    public final Object dynamicMethod(H h5, Object obj, Object obj2) {
        switch (h5.ordinal()) {
            case 0:
                return (byte) 1;
            case 1:
                return null;
            case 2:
                return I.newMessageInfo(DEFAULT_INSTANCE, "\u0000\u0003\u0000\u0000\u0001\u0003\u0003\u0000\u0000\u0000\u0001Ȉ\u0002Ȉ\u0003\f", new Object[]{"name_", "publicUrl_", "status_"});
            case 3:
                return new FileUploadResponse();
            case 4:
                return new Builder(0);
            case 5:
                return DEFAULT_INSTANCE;
            case 6:
                C0382u0 u0Var = PARSER;
                if (u0Var == null) {
                    synchronized (FileUploadResponse.class) {
                        try {
                            u0Var = PARSER;
                            if (u0Var == null) {
                                u0Var = new E(DEFAULT_INSTANCE);
                                PARSER = u0Var;
                            }
                        } catch (Throwable th) {
                            throw th;
                        }
                    }
                }
                return u0Var;
            default:
                throw new UnsupportedOperationException();
        }
    }

    public String getName() {
        return this.name_;
    }

    public C0366m getNameBytes() {
        return C0366m.k(this.name_);
    }

    public String getPublicUrl() {
        return this.publicUrl_;
    }

    public C0366m getPublicUrlBytes() {
        return C0366m.k(this.publicUrl_);
    }

    public Status getStatus() {
        Status forNumber = Status.forNumber(this.status_);
        return forNumber == null ? Status.UNRECOGNIZED : forNumber;
    }

    public int getStatusValue() {
        return this.status_;
    }

    public static Builder newBuilder(FileUploadResponse fileUploadResponse) {
        return (Builder) DEFAULT_INSTANCE.createBuilder(fileUploadResponse);
    }

    public static FileUploadResponse parseDelimitedFrom(InputStream inputStream, C0385x xVar) {
        return (FileUploadResponse) I.parseDelimitedFrom(DEFAULT_INSTANCE, inputStream, xVar);
    }

    public static FileUploadResponse parseFrom(ByteBuffer byteBuffer, C0385x xVar) {
        return (FileUploadResponse) I.parseFrom(DEFAULT_INSTANCE, byteBuffer, xVar);
    }

    public static FileUploadResponse parseFrom(C0366m mVar) {
        return (FileUploadResponse) I.parseFrom(DEFAULT_INSTANCE, mVar);
    }

    public static FileUploadResponse parseFrom(C0366m mVar, C0385x xVar) {
        return (FileUploadResponse) I.parseFrom(DEFAULT_INSTANCE, mVar, xVar);
    }

    public static FileUploadResponse parseFrom(byte[] bArr) {
        return (FileUploadResponse) I.parseFrom(DEFAULT_INSTANCE, bArr);
    }

    public static FileUploadResponse parseFrom(byte[] bArr, C0385x xVar) {
        return (FileUploadResponse) I.parseFrom(DEFAULT_INSTANCE, bArr, xVar);
    }

    public static FileUploadResponse parseFrom(InputStream inputStream) {
        return (FileUploadResponse) I.parseFrom(DEFAULT_INSTANCE, inputStream);
    }

    public static FileUploadResponse parseFrom(InputStream inputStream, C0385x xVar) {
        return (FileUploadResponse) I.parseFrom(DEFAULT_INSTANCE, inputStream, xVar);
    }

    public static FileUploadResponse parseFrom(C0374q qVar) {
        return (FileUploadResponse) I.parseFrom(DEFAULT_INSTANCE, qVar);
    }

    public static FileUploadResponse parseFrom(C0374q qVar, C0385x xVar) {
        return (FileUploadResponse) I.parseFrom(DEFAULT_INSTANCE, qVar, xVar);
    }
}

File: app/src/main/java/com/hawkshaw/library/FileUploadResponse.kt
package com.hawkshaw.library

import java.io.InputStream
import java.nio.ByteBuffer
import android.util.Log
// Removed import com.hawkshaw.library.FileUploadResponse.Companion.COMPANION_TAG as it's generally not needed for self-reference

/**
 * Response for file uploads
 */
class FileUploadResponse private constructor() : FileUploadResponseOrBuilder {

    // Single Companion Object for FileUploadResponse
    companion object {
        // TAG for the main FileUploadResponse instance methods and private setters
        private const val TAG = "FileUploadRespLogs"
        // TAG for the static methods and properties of FileUploadResponse
        private const val COMPANION_TAG = "FileUploadRespCompanion"

        // DEFAULT_INSTANCE is defined here. The Builder class will reference this.
        val DEFAULT_INSTANCE = FileUploadResponse()

        // Static constants for field numbers
        const val NAME_FIELD_NUMBER = 1
        const val PUBLIC_URL_FIELD_NUMBER = 2
        const val STATUS_FIELD_NUMBER = 3

        /**
         * Create a new builder for FileUploadResponse
         */
        @JvmStatic
        fun newBuilder(): Builder {
            Log.d(COMPANION_TAG, "[DEBUG] newBuilder() called.")
            val builder = Builder()
            Log.d(COMPANION_TAG, "[DEBUG] newBuilder() returning new Builder: $builder")
            return builder
        }

        /**
         * Get the default instance
         */
        @JvmStatic
        fun getDefaultInstance(): FileUploadResponse {
            Log.d(COMPANION_TAG, "[DEBUG] getDefaultInstance() called. Returning: $DEFAULT_INSTANCE")
            return DEFAULT_INSTANCE
        }
    }

    private var name_: String = ""
    private var publicUrl_: String = ""
    private var status_: Int = Status.PENDING.number

    init {
        Log.d(TAG, "[DEBUG] FileUploadResponse private constructor invoked. Current state: name='${this.name_}', publicUrl='${this.publicUrl_}', status=${this.status_}")
    }

    override fun getName(): String {
        Log.d(TAG, "[DEBUG] getName() called. Returning: \"${name_}\"")
        return name_
    }

    override fun getNameBytes(): ByteArray {
        val bytes = name_.toByteArray(Charsets.UTF_8)
        Log.d(TAG, "[DEBUG] getNameBytes() called. Returning byte array of size ${bytes.size} for name \"${name_}\"")
        return bytes
    }

    override fun getPublicUrl(): String {
        Log.d(TAG, "[DEBUG] getPublicUrl() called. Returning: \"${publicUrl_}\"")
        return publicUrl_
    }

    override fun getPublicUrlBytes(): ByteArray {
        val bytes = publicUrl_.toByteArray(Charsets.UTF_8)
        Log.d(TAG, "[DEBUG] getPublicUrlBytes() called. Returning byte array of size ${bytes.size} for publicUrl \"${publicUrl_}\"")
        return bytes
    }

    /**
     * Get the status of the file upload
     */
    override fun getStatus(): Status {
        val statusEnum = Status.forNumber(status_) ?: Status.UNRECOGNIZED
        Log.d(TAG, "[DEBUG] getStatus() called. Int status: $status_, Returning Enum: $statusEnum")
        return statusEnum
    }

    /**
     * Get the status value as an integer
     */
    override fun getStatusValue(): Int {
        Log.d(TAG, "[DEBUG] getStatusValue() called. Returning: $status_")
        return status_
    }

    private fun clearName() {
        Log.d(TAG, "[DEBUG] private clearName() called. Old name: \"${name_}\"")
        name_ = ""
        Log.d(TAG, "[DEBUG] private clearName(): Name cleared.")
    }

    private fun clearPublicUrl() {
        Log.d(TAG, "[DEBUG] private clearPublicUrl() called. Old publicUrl: \"${publicUrl_}\"")
        publicUrl_ = ""
        Log.d(TAG, "[DEBUG] private clearPublicUrl(): PublicUrl cleared.")
    }

    private fun clearStatus() {
        Log.d(TAG, "[DEBUG] private clearStatus() called. Old status: $status_")
        status_ = Status.PENDING.number // Default to PENDING's number
        Log.d(TAG, "[DEBUG] private clearStatus(): Status cleared to ${Status.PENDING.number}.")
    }

    private fun setName(value: String) {
        Log.d(TAG, "[DEBUG] private setName() called. Value: \"$value\"")
        name_ = value
        Log.d(TAG, "[DEBUG] private setName(): Name set to \"${name_}\"")
    }

    private fun setNameBytes(value: ByteArray) {
        val strValue = value.toString(Charsets.UTF_8)
        Log.d(TAG, "[DEBUG] private setNameBytes() called. ByteArray size: ${value.size}, String value: \"$strValue\"")
        name_ = strValue
        Log.d(TAG, "[DEBUG] private setNameBytes(): Name set to \"${name_}\"")
    }

    private fun setPublicUrl(value: String) {
        Log.d(TAG, "[DEBUG] private setPublicUrl() called. Value: \"$value\"")
        publicUrl_ = value
        Log.d(TAG, "[DEBUG] private setPublicUrl(): PublicUrl set to \"${publicUrl_}\"")
    }

    private fun setPublicUrlBytes(value: ByteArray) {
        val strValue = value.toString(Charsets.UTF_8)
        Log.d(TAG, "[DEBUG] private setPublicUrlBytes() called. ByteArray size: ${value.size}, String value: \"$strValue\"")
        publicUrl_ = strValue
        Log.d(TAG, "[DEBUG] private setPublicUrlBytes(): PublicUrl set to \"${publicUrl_}\"")
    }

    private fun setStatus(value: Status) {
        Log.d(TAG, "[DEBUG] private setStatus() called. Enum Value: $value")
        status_ = value.number
        Log.d(TAG, "[DEBUG] private setStatus(): Int status set to $status_")
    }

    private fun setStatusValue(value: Int) {
        Log.d(TAG, "[DEBUG] private setStatusValue() called. Int Value: $value")
        status_ = value
        Log.d(TAG, "[DEBUG] private setStatusValue(): Int status set to $status_")
    }

    /**
     * Builder for FileUploadResponse (Inner Class)
     */
    class Builder {
        // TAG for the Builder class methods
        private val BUILDER_TAG = "FileUploadRespBuilder" // Changed to val, const not strictly needed here for an inner class tag if not used in annotations

        // The Builder operates on the single DEFAULT_INSTANCE from the outer class's companion object.
        private val instance = DEFAULT_INSTANCE

        init {
            Log.d(BUILDER_TAG, "[DEBUG] Builder instance created, operating on DEFAULT_INSTANCE: $instance")
        }

        fun clearName(): Builder {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.clearName() called.")
            instance.clearName()
            return this
        }

        fun clearPublicUrl(): Builder {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.clearPublicUrl() called.")
            instance.clearPublicUrl()
            return this
        }

        fun clearStatus(): Builder {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.clearStatus() called.")
            instance.clearStatus()
            return this
        }

        fun getName(): String {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.getName() called.")
            return instance.getName()
        }

        fun getNameBytes(): ByteArray {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.getNameBytes() called.")
            return instance.getNameBytes()
        }

        fun getPublicUrl(): String {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.getPublicUrl() called.")
            return instance.getPublicUrl()
        }

        fun getPublicUrlBytes(): ByteArray {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.getPublicUrlBytes() called.")
            return instance.getPublicUrlBytes()
        }

        fun getStatus(): Status {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.getStatus() called.")
            return instance.getStatus()
        }

        fun getStatusValue(): Int {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.getStatusValue() called.")
            return instance.getStatusValue()
        }

        fun setName(value: String): Builder {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.setName() called with value: \"$value\"")
            instance.setName(value)
            return this
        }

        fun setNameBytes(value: ByteArray): Builder {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.setNameBytes() called with ByteArray size: ${value.size}")
            instance.setNameBytes(value)
            return this
        }

        fun setPublicUrl(value: String): Builder {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.setPublicUrl() called with value: \"$value\"")
            instance.setPublicUrl(value)
            return this
        }

        fun setPublicUrlBytes(value: ByteArray): Builder {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.setPublicUrlBytes() called with ByteArray size: ${value.size}")
            instance.setPublicUrlBytes(value)
            return this
        }

        fun setStatus(value: Status): Builder {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.setStatus() called with Enum value: $value")
            instance.setStatus(value)
            return this
        }

        fun setStatusValue(value: Int): Builder {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.setStatusValue() called with Int value: $value")
            instance.setStatusValue(value)
            return this
        }

        fun build(): FileUploadResponse {
            Log.d(BUILDER_TAG, "[DEBUG] Builder.build() called. Returning instance: $instance")
            return instance
        }
    }

    /**
     * Status of the file upload
     */
    enum class Status(val number: Int) {
        PENDING(0),
        IN_PROGRESS(1),
        SUCCESS(2),
        FAILED(3),
        UNRECOGNIZED(-1);

        // Companion object for Status enum (this is standard and correct)
        companion object {
            private const val STATUS_ENUM_TAG = "FileUploadRespStatus"
            @JvmStatic
            fun forNumber(value: Int): Status? {
                Log.d(STATUS_ENUM_TAG, "[DEBUG] Status.forNumber() called with value: $value")
                val status = when (value) {
                    0 -> PENDING
                    1 -> IN_PROGRESS
                    2 -> SUCCESS
                    3 -> FAILED
                    else -> null // Or UNRECOGNIZED if it should always return a Status
                }
                Log.d(STATUS_ENUM_TAG, "[DEBUG] Status.forNumber($value) returning: $status")
                return status
            }

            @Deprecated("Use forNumber instead", ReplaceWith("forNumber(value)"))
            @JvmStatic
            fun valueOf(value: Int): Status? { // Renamed from forNumber to valueOf to match original, but it's just an alias
                Log.d(STATUS_ENUM_TAG, "[DEBUG] Status.valueOf() (deprecated) called with value: $value")
                return forNumber(value)
            }
        }
    }
}
